pinot-spi/src/main/java/org/apache/pinot/spi/config/user/UserConfig.java

```diff
@@ -0,0 +1,143 @@
+    private ComponentType _componentType;
+
+    @JsonPropertyDescription("The role of user")
+    private RoleType _roleType;
+
+    @JsonPropertyDescription("The tables owned of User")
+    private List<String> _tables;
+
+    @JsonPropertyDescription("The table permission of User")
+    private List<AccessType> _permissions;
+
+    @JsonCreator
+    public UserConfig(@JsonProperty(value = USERNAME_KEY, required = true) String username,
+        @JsonProperty(value = PASSWORD_KEY, required = true) String password,
+        @JsonProperty(value = COMPONET_KEY, required = true) String component,
+        @JsonProperty(value = ROLE_KEY, required = true) String role,
+        @JsonProperty(value = TABLES_KEY) @Nullable List<String> tableList,
+        @JsonProperty(value = PERMISSIONS_KEY) @Nullable List<AccessType> permissionList
+    ) {
+        Preconditions.checkArgument(username != null, "'username' must be configured");
+        Preconditions.checkArgument(password != null, "'password' must be configured");
+
+        // NOTE: Handle lower case table type and raw table name for backward-compatibility
+        _username = username;
+        _password = password;
+        _componentType = ComponentType.valueOf(component.toUpperCase());
+        _roleType = RoleType.valueOf(role.toUpperCase());
+        _tables = tableList;
+        _permissions = permissionList;
+    }
+
+    @JsonProperty(USERNAME_KEY)
+    public String getUserName() {
+        return _username;
+    }
+
+    public String getUsernameWithComponent() {
+        return getUserName() + "_" + getComponentType().toString();
+    }
+
+    public boolean isExist(String username, ComponentType component) {
+        return _username.equals(username) && _componentType.equals(component);
+    }
+
+    @JsonProperty(PASSWORD_KEY)
+    public String getPassword() {
+        return _password;
+    }
+
+    @JsonProperty(TABLES_KEY)
+    public List<String> getTables() {
+        return _tables;
+    }
+
+    @JsonProperty(PERMISSIONS_KEY)
+    public List<AccessType> getPermissios() {
+        return _permissions;
+    }
+
+    @JsonProperty(COMPONET_KEY)
+    public ComponentType getComponentType() {
+        return _componentType;
+    }
+
+    @JsonProperty(ROLE_KEY)
+    public RoleType getRoleType() {
+        return _roleType;
+    }
+
+    public void setRole(String roleTypeStr) {
+        _roleType = RoleType.valueOf(roleTypeStr);
+    }
+
+    public void setPassword(String password) {
+        _password = password;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        if (!super.equals(o)) {
+            return false;
+        }
+        UserConfig that = (UserConfig) o;
+        return _username.equals(that._username) && _componentType == that._componentType;
+    }
```

Comment:
this might lead to unexpected behavior. what's the rationale for using `username` and `component` only?

Reply:
@apucher Thanks for your considerable consider, but in our implementation, this Class UserConfig represents Auth User Entity that is stored in ZK. And both username and component fields could enough identify unique user entity.